lib/src/signal_client.dart

```diff
@@ -48,31 +38,77 @@ class SignalClient {
 
   bool get connected => _connected;
 
-  Future<void> join(String url, String token, JoinOptions? options) async {
-    final rtcUrl = '$url/rtc';
-    var params = _joinParams(token);
-    if (options != null && options.autoSubscribe != null) {
-      params += '&auto_subscribe=${options.autoSubscribe! ? '1' : '0'}';
-    }
+  Uri _buildUri(
+    String uriString, {
+    required String token,
+    ConnectOptions? options,
+    bool reconnect = false,
+    bool validate = false,
+  }) {
+    final uri = Uri.parse(uriString);
+
+    return uri.replace(
+      //
+      // It is possible to enforce WSS
+      //
+      scheme: validate ? 'http' : uri.scheme,
```

Comment:
this would break if the server is running behind TLS. in validate mode, it'll be https if original url was wss. Otherwise it'll be http / ws

Reply:
Yup I'll fix that to match wss/https and ws/http